src/components/EtherspotContractTransaction.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useContext, useEffect, useId } from 'react';
+import { ethers } from 'ethers';
+
+// contexts
+import EtherspotBatchContext from '../contexts/EtherspotBatchContext';
+
+// types
+import { IEtherspotContractTransaction } from '../types/EtherspotUi';
+
+interface EtherspotContractTransactionProps extends IEtherspotContractTransaction {
+  children?: React.ReactNode;
+}
+
+const EtherspotContractTransaction = ({
+  children,
+  value,
+  contractAddress,
+  params,
+  methodName,
+  abi,
+  id: transactionId
+}: EtherspotContractTransactionProps) => {
+  const context = useContext(EtherspotBatchContext);
+  const componentId = useId();
+
+  if (context === null) {
+    throw new Error('No parent <EtherspotBatch />');
+  }
+
+  let contractInterface;
+  try {
+    contractInterface = new ethers.utils.Interface(abi);
+  } catch (e) {
+    if (e instanceof Error && e?.message) {
+      throw new Error(`Failed to build contract interface from provided ABI: ${e.message}`);
```

Comment:
Remember to suggest what should the user do next if they encounter this error.

Reply:
the error here comes explanatory from `ethers.js`, however I can add some directions in case there's no error and no contract/abi built (few lines below)

Reply:
added on both actually